backend/store/etcd/watcher.go

```diff
@@ -87,35 +86,30 @@ func (w *Watcher) start(ctx context.Context) {
 		defer close(w.resultChan)
 		_ = limiter.Wait(ctx)
 		for ctx.Err() == nil {
-			for watchResponse := range watcherChan {
+			select {
+			case watchResponse, closed := <-watcherChan:
```

Comment:
It's unclear to me why this variable is named `closed` - a true value here indicates that the value was received from an open channel.

Reply:
It shouldn't be named closed, you're 100% correct. It should be named `ok`.